test/Models/RegionProviderListSpec.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+/*global require,describe,it,expect,beforeEach,fail*/
+
+var Terria = require('../../lib/Models/Terria');
+var RegionProviderList = require('../../lib/Models/RegionProviderList');
+var RegionProvider = require('../../lib/Models/RegionProvider');
+var DataTable = require('../../lib/Map/DataTable.js');
+var terria;
+var rpl;
+
+beforeEach(function() {
+    terria = new Terria({
+        baseUrl: './',
+        regionMappingDefinitionsUrl: 'test/csv/regionMapping.json',
+    });
+    terria.corsProxy.baseProxyUrl = ""; // there is no localhost:3002/proxy, so ...
+    rpl = new RegionProviderList(terria);
+});
+
+describe('RegionProviderList', function() {
+  it('loads some region providers', function(done) {
+    rpl.init().then(function(x) {
+      expect(rpl.regionProviders.length).toBeGreaterThan(2);
+    }).otherwise(fail).then(done);
+  }); 
+  /*
+  Test fails...and do we care anyway?
+  it('does not allow duplicate identifiers', function(done) {
```

Comment:
apparently `JSON.parse` doesn't fail if the JSON string contains duplicate keys.  But if we change to using an array instead of an object in order to preserve order, it could be worthwhile (but not essential) to check for duplicate IDs.